src/main/scala/fi/aalto/cs/apluscourses/intellij/actions/ReplAction.scala

```diff
@@ -45,25 +61,49 @@ class ReplAction extends RunConsoleAction {
     val selectedModule = getScalaModuleOfEditorFile(project, dataContext)
       .orElse(getScalaModuleOfSelectedFile(project, dataContext))
 
-    val setting = settings.headOption.getOrElse {
-      val factory = configurationType.getConfigurationFactories.head
-      val configurationName = selectedModule
-        .map(module => s"REPL for ${module.getName}")
-        .getOrElse("Scala REPL")
-      runManagerEx.createConfiguration(configurationName, factory)
-    }
+    val setting = runManagerEx.createConfiguration("Scala REPL", new ReplConfigurationFactory())
     val configuration = setting.getConfiguration.asInstanceOf[ScalaConsoleRunConfiguration]
 
     selectedModule match {
       case Some(module) =>
-        if (setConfigurationConditionally(project, module, configuration)) {
-          RunConsoleAction.runExisting(setting, runManagerEx, project)
+        if (!setConfigurationConditionally(project, module, configuration)) {
+          return // scalastyle:ignore
         }
-      case None =>
-        // For now, no dialog is shown for a project level REPL
-        configuration.setName("Scala REPL")
-        RunConsoleAction.runExisting(setting, runManagerEx, project)
+      case None => // For now, no dialog is shown for a project level REPL
     }
+    RunConsoleAction.runExisting(setting, runManagerEx, project)
+  }
+
+  private class ReplConfigurationFactory() extends ConfigurationFactory(getMyConfigurationType) {
```

Comment:
To get rid of a warning pointed out by @nikke234 [on Slack](https://apluslms.slack.com/archives/CSDV9CA31/p1595961149005200), this class should override `getId` so that it always returns the same string, for example, `"A+ extended Scala REPL"`.

Reply:
aaaaaaaand, solved, thanks to the handsome gentleman from the Eastside! :kissing_heart: